ledger/persistedaccts_list.go

```diff
@@ -0,0 +1,127 @@
+package ledger
+
+// persistedAccountDataList represents a doubly linked list.
+// must initiate with newPersistedAccountList.
+type persistedAccountDataList struct {
+	root     persistedAccountDataListNode // sentinel list element, only &root, root.prev, and root.next are used
+	len      int                          // current list length excluding (this) sentinel element
+	freeList *persistedAccountDataList    // preallocated nodes location
+}
+
+type persistedAccountDataListNode struct {
+	// Next and previous pointers in the doubly-linked list of elements.
+	// To simplify the implementation, internally a list l is implemented
+	// as a ring, such that &l.root is both the next element of the last
+	// list element (l.Back()) and the previous element of the first list
+	// element (l.Front()).
+	next, prev *persistedAccountDataListNode
+
+	// The list to which this element belongs. (helps removing items fast)
+	list *persistedAccountDataList
```

Comment:
maintaining a `list *persistedAccountDataList` doesn't get you much - other than validation. 
I think you'd be better off without that extra validation - since it's not really an issue anyway for a correctly written code.